openassessment/tests/test_workflow_batch_update_api.py

```diff
@@ -0,0 +1,545 @@
+        # we expect 0 blocked submissions for "course_2"
+        self.assertEqual(len(blocked), 0)
+
+    def test_get_blocked_peer_workflows_for_ora_block(self):
+        # pylint: disable=unused-variable
+        tim_sub, tim = self._create_student_and_submission("Tim", "Tim's answer")
+        # pylint: disable=unused-variable
+        miles_sub, miles = self._create_student_and_submission("Miles", "Miles's answer")
+        # pylint: disable=unused-variable
+        pat_sub, pat = self._create_student_and_submission("Pat", "Pat's answer")
+
+        blocked = update_api.get_blocked_peer_workflows(item_id="item_1")
+        # we expect 0 blocked submissions for "item_1"
+        self.assertEqual(len(blocked), 0)
+
+        pw_tim = PeerWorkflow.objects.get(student_id=tim["student_id"])
+        pw_tim.created_at = timezone.now() - datetime.timedelta(days=8)
+        pw_tim.save()
+
+        pw_pat = PeerWorkflow.objects.get(student_id=pat["student_id"])
+        pw_pat.created_at = timezone.now() - datetime.timedelta(days=8)
+        pw_pat.item_id = "item_1"
+        pw_pat.save()
+
+        blocked = update_api.get_blocked_peer_workflows(item_id="item_1")
+        # we expect 1 blocked submission for "item_1"
+        self.assertEqual(len(blocked), 1)
+
+        pw_miles = PeerWorkflow.objects.get(student_id=miles["student_id"])
+        pw_miles.item_id = "item_2"
+        pw_miles.save()
+
+        blocked = update_api.get_blocked_peer_workflows(item_id="item_2")
+        # we expect 0 blocked submissions for "item_2"
+        self.assertEqual(len(blocked), 0)
+
+    def test_is_flexible_peer_grading_on(self):
+        workflow_requirements = {"peer": {
+            "must_grade": 2,
+            "must_be_graded_by": 3,
+            "enable_flexible_grading": True
+        }}
+        ora_block = MockOraBlock(workflow_requirements)
+        course_block = MockCourseBlock(False)
+        self.assertTrue(update_api.is_flexible_peer_grading_on(ora_block, course_block))
+
+        ora_block.requirements['peer']['enable_flexible_grading'] = False
+        self.assertFalse(update_api.is_flexible_peer_grading_on(ora_block, course_block))
+
+        ora_block.requirements['peer']['enable_flexible_grading'] = False
+        self.assertFalse(update_api.is_flexible_peer_grading_on(ora_block, course_block))
+
+        del (ora_block.requirements['peer']['enable_flexible_grading'])
+        self.assertFalse(update_api.is_flexible_peer_grading_on(ora_block, course_block))
+
+        del (ora_block.requirements['peer'])
+        self.assertFalse(update_api.is_flexible_peer_grading_on(ora_block, course_block))
+
+        course_block.force_on_flexible_peer_openassessments = True
+        self.assertTrue(update_api.is_flexible_peer_grading_on(ora_block, course_block))
+
+    @patch('openassessment.workflow_batch_update_api.modulestore')
+    @patch('openassessment.workflow_batch_update_api.UsageKey.from_string')
```

Comment:
nit: you could avoid this by just using valid course keys. course-v1:org+course+1 course-v1:org+course+2